src/net.cpp

```diff
@@ -906,36 +906,50 @@ std::pair<size_t, bool> CConnman::SocketSendData(CNode& node) const
 {
     auto it = node.vSendMsg.begin();
     size_t nSentSize = 0;
-
-    while (it != node.vSendMsg.end()) {
-        const auto& data = *it;
-        assert(data.size() > node.nSendOffset);
+    bool have_unsent_data{false};
```

Comment:
8d1a91f6394f00b880613268a9f2f3164ef914e2: `data_left` is more consistent with the header

Reply:
Or perhaps something like `return {nSentSize, socket_empty && message_queue_empty};`